### ISSUE TITLE
EDV documents can be or should be Unbound VCs

### ISSUE THREAD

**Issue by mwherman2000** [decentralized-identity/edv-spec#65]
Checkout the last few minutes of https://youtu.be/FFv4WZ0p3aY for a definition of an Unbound VC.

**Comment by TallTed** [comment]
@mwherman2000 -- It would be helpful if you could summarize that definition here, in text and/or image(s), and provide some attendant discussion of how your issue title (which is not [yet, at least] a resolution; is it a proposal? something else?) flows from there...

**Comment by mwherman2000** [comment]
A PR to the VC data model specification describes Bound Credentials and Unbound Credentials: 
https://github.com/w3c/vc-data-model/pull/788

**Comment by mwherman2000** [comment]
![image](https://user-images.githubusercontent.com/6101736/128497095-e16a126a-488d-448a-bdf8-25dfd8bae50e.png)

**Comment by agropper** [comment]
Why MUST a bound credential be issued to the Holder?

A (bound) credential is verifiable on its own and has value because some
claims MAY identify the subject independently of any particular Holder.

**Comment by mwherman2000** [comment]
When a (bound) credential that has a credentialSubject id set equal to the subject of the credential issued by an Issuer, it's primary purpose (raison d'etre) to be held by the subject Holder for the subsequence creation of a verifiable presentation.

I don't if a bound credential MUST be issued to the Holder.

@swcurran Can you help me out here?

**Comment by swcurran** [comment]
The concept of bound and unbound VCs stems for a world view difference about the participants in the VC trust triangle, and that I (at least) think exposes a crucial weakness in the VC 1.0 Data Model. We think the bound/unbound concept is a way to allow for both views to be supported without a radical change to the spec.

Background: As we in Aries have moved from using only Indy AnonCreds to adding support for the W3C standard, we've discovered a huge difference in the cryptographic trust model between the two. In the AnonCreds model, the holder is a cryptographic component of the Trust Triangle, while in the W3C VC model, it is not. In AnonCreds, the Issuer MUST include in the verifiable credential a proof of the Holder on issueance, and the verifier MUST verify the verifiable presentation comes from the Holder. In the VC model, there is no such required binding to the Holder -- anyone can hold an issued VC. You can bind it to the holder (and we think many would) but it is not required and in fact, barely discussed in the spec.

Adding the concept of Bound and Unbound VCs could enable support for both models.

- Unbound: The Issuer issued a claim that the verifier can verify. "Alice said the Sky is Blue"
- Bound: The Issuer made a claim to the Holder that the verifier can verify the claim and that it was made to the holder. "Alice said to Carol that the Sky is Blue"

Of course, while the cryptographic trust is specified in both models, the verify must still make their human trust decision -- do they trust the issuer?  However, the Bound model also allows the verifier to make a decision about whether they trust the holder.

[section 5.1](https://www.w3.org/TR/vc-data-model/#lifecycle-details) of the W3C V3 Spec. is quite jarring for those that assume the Holder is a core participant in the model. That a VC can be transfered freely amongst holders is, well, quite stunning to those of us with that view.

Basically, all of our use cases (and all that we can think of) **require** the concept that the Issuer issues the verifiable credential to the Holder and only that Holder can present a proof based on that verifiable credential. We see far less value for credentials that do not include the holder as a cryptographic participant.

**Comment by swcurran** [comment]
@mwherman2000 -- in reviewing the VC spec. more thoroughly, I would alter your definition slightly. I think "Bound" should mean that the VC was issued to the holder identified in the VC -- irrespective of the subject. For a Bound VC, the verifier should require proof from the holder. That proof could be an explicit binding to the identifier (e.g. proof of control over a DID), or a ZKP of the binding between the VC and holder.

The data in a Bound VC can (of course) be about the holder or about another subject. That is determined by the meaning of the claims in the VC, independent of the binding to the holder.

**Comment by mwherman2000** [comment]
>  I think "Bound" should mean that the VC was issued to the holder identified in the VC -- irrespective of the subject. 

Can we make this criterion "the holder identified in the VC" more concrete? Which specific element in the VC data model are we referring to?  For example: ...the top-level credential id? ...the credentialSubject id? ...something else?

**Comment by tplooker** [comment]
I dont think bound and un-bound VC's is actually the best way to describe what is being discussed here. If a VC does not feature a cryptographic binding to the holder on issuance, it is not necessarily "un-bound", it just doesn't use a cryptographic holder binding. For example other forms of valid binding mechanisms that are present across the digital credential space are:

1. Bearer based credentials - Essentially only someone with the credential can exercise its capabilities, best example is a concert ticket, easily transferable by design but can suffer from replay attacks depending on the cryptographic scheme used.
2. Attribute bound credentials - Example is issuing a credential and including a personal attribute(s) about the person, such as their name that enables authentication when the credential is presented in conjunction with another credential featuring a trusted authenticator that shares the same attribute(s). An example of this is a paper based health credential that features a persons first and last name who's presentation must be made along side valid photo ID (e.g drivers license).

**Comment by mwherman2000** [comment]
> I dont think bound and un-bound VC's is actually the best way to describe what is being discussed here. 

@tplooker Are you challenging the 2 concepts, Bound and Unbound Credentials, from only a naming/labeling perspective? If so, can you suggest a different pair of names/labels?
....or are you challenging the entire underlying basis for the 2 concepts?

**Comment by dlongley** [comment]
Documents stored in an EDV may have nothing to do with verifiable credentials. I think EDVs should be able to store arbitrary structured data as this makes the fewest assumptions about use.

The issue title seems to imply that because the VC format exists that we should try to model *all* data as VCs (or perhaps just all documents for new specs?). If so, I think that is unduly reversing the burden. I think if we wanted to limit what EDVs can store to VCs then we'd need a good reason to create that limitation. Perhaps I misunderstand the aim.

**Comment by mwherman2000** [comment]
@dlongley In effect, I'm advocating that every EDV resource/document have a common standard exterior/wrapper (in fact, every Fully Decentralized Object (FDO) on the planet) modeled after Unbound VCs.  

There are, in effect, 2 subproposals here:
a. that every EDV resource/document have a common standard exterior/wrapper
b. that the common standard exterior/wrapper be modeled after Unbound VCs

>>I think EDVs should be able to store arbitrary structured data

I agree. Any form of structured data can be stored as an element in the credentialSubject of the EDV resource wrapper.

**Comment by dlongley** [comment]
Why would we want to restrict EDV documents to be VCs? Certainly, people *can* use VCs if they want to, by why would we prevent people from being able to store non-VCs?

**Comment by mwherman2000** [comment]
> Why would we want to restrict EDV documents to be VCs? Certainly, people _can_ use VCs if they want to, by why would we prevent people from being able to store non-VCs?

I'm not at all suggesting that EDV documents be restricted to actual VCs.  "VC" is legacy terminology. I'm going to change all references of "Unbound VC" to "Unbound Credentials" ...as a small step forward.

See https://github.com/w3c/vc-data-model/pull/788#issuecomment-895251315

**Comment by dlongley** [comment]
> "VC" is legacy terminology.

I don't agree or maybe I don't understand this claim. But it may not be important to the conversation.

> I'm not at all suggesting that EDV documents be restricted to actual VCs. "VC" is legacy terminology. I'm going to change all references of "Unbound VC" to "Unbound Credentials" ...as a small step forward.

Ok, then my question just becomes, why would we prevent people from being able to store non-"Unbound Credentials"? But I don't really know what "Unbound Credentials" are if they are not VCs. But perhaps that's not important (at least not just yet).

If I understand the proposal here, it is to require everyone who wants to store structured data in an EDV to represent their data as an "Unbound Credential", which, presumably, brings along a number of important constraints. Why would we impose these constraints (whatever they are) on everyone who wants to store something in an EDV?

**Comment by mwherman2000** [comment]
>If I understand the proposal here, it is to require everyone who wants to store structured data in an EDV to represent their data as an "Unbound Credential", which, presumably, brings along a number of important constraints. Why would we impose these constraints (whatever they are) on everyone who wants to store something in an EDV?

@dlongley Really? You're assuming there are a bunch of constraints (that don't exist) and then use these unfounded (unbound?) assumptions to reject the idea. #jeepers

**Comment by dlongley** [comment]
> You're assuming there are a bunch of constraints (that don't exist) and then further assume these unfounded (unbound?) assumptions are going to impose further constraints "whatever they are" on everyone.

If there are no constraints, what does it mean to say "EDV documents should be Unbound Credentials" (from the issue title)? You didn't mean "EDV documents should have an unconstrained format", did you? Presumably one can tell whether a document is an "Unbound credential" or not by checking some constraints. If not, "Unbound credential" is not a meaningful term in my view.  I don't think that's a strong assumption for any categorization. So, if there are such constraints, why would we impose them on EDV documents?

I think my question is a simple one -- and one that should be asked for any potential decision of this nature. If we're going to say that "EDV documents should be X", then we should have a good reason for that and we should understand that if they "should be X" then they necessarily "should not be not-X" -- and what the consequences of that proposal are.

**Comment by dlongley** [comment]
@mwherman2000,

If it helps, you clearly outlined some of the constraints for "Unbound Credentials" in this very thread (see [this comment]( https://github.com/decentralized-identity/edv-spec/issues/65#issuecomment-894165917)), so I don't think it should be controversial to say they exist. For example, they must (?) have an `@context` and `proof`. So, why would we require every EDV document to have these things? This is the question I'm asking.

**Comment by mwherman2000** [comment]
@dlongley touche' :-) I was refering to higher-level, more potentially consequential constraints. :-)

In my actual implementation of Unbound Credentials (Structured Credentials), `trustLevel` is used to discern what proof, if any, is required for a particular credential ...in effect, making the proof optional ...along a sliding scale 
(See https://github.com/mwherman2000/TrustedDigitalWeb/blob/master/TDW.Servers/TDW.TRACommon/TRACommon.tsl#L9).

My proposal remains the same - but I need to further refine my "proof" :-)

**Comment by dmitrizagidulin** [comment]
Discussed on the Sep 2, 2021 call.

@mwherman2000 - What is the use case / requirement for switching to the format you're proposing?

Propose add a 'Close in 14 days' tag on this issue on the next call.

**Comment by dmitrizagidulin** [comment]
@mwherman2000 - just checking in, to make sure there's no objection to closing the issue.
